src/pylibsshext/channel.pyx

```diff
@@ -25,7 +25,52 @@ from libc.string cimport memcpy, memset
 
 from pylibsshext.session cimport get_libssh_session
 from pylibsshext.errors cimport LibsshChannelException
-from subprocess import CalledProcessError, CompletedProcess
+from subprocess import CalledProcessError
+
+
+# Defined CompletedProcess here to be compliant with py2
+class CompletedProcess(object):
+    """A process that has finished running.
+    This is returned by run().
+    Attributes:
+      args: The list or str args passed to run().
+      returncode: The exit code of the process, negative for signals.
+      stdout: The standard output (None if not captured).
+      stderr: The standard error (None if not captured).
+    """
+    def __init__(self, args, returncode, stdout=None, stderr=None):
+        self.args = args
+        self.returncode = returncode
+        self.stdout = stdout
+        self.stderr = stderr
+
+    def __repr__(self):
+        args = ['args={!r}'.format(self.args),
+                'returncode={!r}'.format(self.returncode)]
+        if self.stdout is not None:
+            args.append('stdout={!r}'.format(self.stdout))
+        if self.stderr is not None:
+            args.append('stderr={!r}'.format(self.stderr))
+        return "{}({})".format(type(self).__name__, ', '.join(args))
+
+
+    def check_returncode(self):
+        """Raise CalledProcessError if the exit code is non-zero."""
+        if self.returncode:
+            raise CalledProcessError(self.returncode, self.args, self.stdout,
+                                     self.stderr)
+
+
+class BufferManager():
+    def __init__(self, filename, mode):
```

Comment:
This wouldn't work because you're not passing the args. OTOH this class is unnecessary anyway.